tools/testing/target_determination/determinator.py

```diff
@@ -19,10 +19,15 @@ def get_test_prioritizations(
         print(f"  {test}", file=file)
 
     for heuristic in HEURISTICS:
-        new_rankings: TestPrioritizations = heuristic.get_prediction_confidence(tests)
-        aggregated_results.add_heuristic_results(heuristic, new_rankings)
+        try:
+            new_rankings: TestPrioritizations = heuristic.get_prediction_confidence(
+                tests
+            )
+            aggregated_results.add_heuristic_results(heuristic, new_rankings)
 
-        print(f"Results from {heuristic.__class__.__name__}")
-        print(new_rankings.get_info_str(verbose=False), file=file)
+            print(f"Results from {heuristic.__class__.__name__}")
+            print(new_rankings.get_info_str(verbose=False), file=file)
+        except Exception as e:
+            print(f"Error in {heuristic.__class__.__name__}: {e}", file=file)
```

Comment:
don't you want to add the original tests here instead of nothing? Cause now in case of an exception it still doesn't add  anything to aggregated_results

Reply:
They all get added later at https://github.com/pytorch/pytorch/pull/138087/files#diff-e72503c9e3e8766e2d1bacf3fad7b88aa166e0e90a7e103e7df99357a35df8d7R1773